doc/conf.py

```diff
@@ -394,3 +400,12 @@ def setup(app):
     breathe_projects["CCF"] = str(doc_dir / breathe_projects["CCF"])
     if not os.environ.get("SKIP_DOXYGEN"):
         subprocess.run(["doxygen"], cwd=root_dir, check=True)
+
+    # configuration generator
+    input_file_path = doc_dir / "host_config_schema/cchost_config.json"
+    output_file_path = doc_dir / "operations/generated_config.rst"
+
+    if os.path.exists(input_file_path):
+        generate_config_rst.generate_configuration_docs(
```

Comment:
Is there a reason we can't implement this as roles/directives (like the typedoc roles above), so that we could reference/link to specific subsections of the config, or paste sections of it rather than the whole doc? Non-critical - obviously generating a single doc works for now. Just wondering if that's something we could do in future.

Reply:
Agreed it would be nice, but non-essential for a first pass.

Reply:
Yes, I think this would be nice too. I was aiming for parity with what the manually-written configuration doc currently looked like, but we can definitely improve the rendering script in the future.